baccmod/fitting.py

```diff
@@ -0,0 +1,161 @@
+                Integer counts array of N dimension
+            exposure_correction: np.array
+                Floating point value to correct for difference of exposure in the data
+            maxiter : int
+                maximum number of iteration for fitting, as the fitting is performed in two steps, could be the double of this value in practice
+
+        Returns
+        -------
+            model : astropy.modeling.Model
+                the fitted model
+        """
+        # work on a copy
+        model_copy = model.copy()
+
+        if exposure_correction is None:
+            exposure_correction = np.ones_like(data, dtype=np.float64)
+
+        # flatten coords & data
+        flat_coords = [c.ravel() for c in coords]
+        flat_data   = data.ravel().astype(int)
+        flat_exposure_correction = exposure_correction.ravel()
+
+        # precompute log‑factorial
+        log_fact = self._log_factorial(flat_data)
+
+        # gather constraint info
+        tied = model_copy.tied or {}
+
+        # build list of free parameters
+        all_params = list(model_copy.param_names)
+        free_params = [p for p in all_params if (p not in tied) or (not tied[p])]
+
+        # initial seeds, bounds, fixed flags
+        seeds  = {p: getattr(model_copy, p).value for p in free_params}
+        bounds = {p: model_copy.bounds.get(p, (None, None)) for p in free_params}
+        fixed  = {p: model_copy.fixed.get(p, False)         for p in free_params}
+
+        # helper to apply parameters and tied relation
+        def apply_params_and_tied(pars):
+            # update free params
+            for name, val in pars.items():
+                setattr(model_copy, name, val)
+            # update tied params
+            for name, rule in tied.items():
+                if callable(rule):
+                    val = rule(model_copy)
+                elif isinstance(rule, str):
+                    ctx = {p: getattr(model_copy, p).value for p in all_params}
+                    val = eval(rule, {}, ctx)
+                else:
+                    continue
+                setattr(model_copy, name, val)
+
+        # negative log‑likelihood
+        def neg_logL(**pars):
+            apply_params_and_tied(pars)
+            mu = model_copy(*flat_coords) * flat_exposure_correction
+            return -np.sum(self._log_poisson(mu, flat_data, log_fact))
+
+        # wrapper to accept both positional and keyword args
+        def fcn_wrapper(*args, **kwargs):
+            if args:
+                pars = dict(zip(free_params, args))
+            else:
+                pars = kwargs
+            return neg_logL(**pars)
+
+        # set up Minuit
+        m = Minuit(fcn_wrapper, name=free_params, **seeds)
+        m.errordef = Minuit.LIKELIHOOD
+
+        # apply bounds & fixed
+        for p in free_params:
+            lo, hi = bounds[p]
+            if fixed[p]:
+                m.fixed[p] = True
+            elif lo is not None or hi is not None:
```

Comment:
Is the elif needed?

Reply:
Could be an if, but should not change anything as if the value is fixed we will not really look at the limits.

Reply:
I was thinking it could be nothing, simply always update the bounds.

Maybe even remove line 122 and use `m.limits[p] = bounds[p]` instead